File: csharp/CodexFeatures/GhostTextManager.cs
namespace TormentNexus.CodexFeatures
{
    public class GhostTextManager
    {
        public string Suggest(string input, int cursorPosition)
        {
            if (input.StartsWith("git c")) return "ommit -m \"update\"";
            if (input.StartsWith("dotnet r")) return "un";
            return string.Empty;
        }

        public string Render(string input, string suggestion)
        {
            return $"{input}\x1b[90m{suggestion}\x1b[0m";
        }
    }
}

File: java/src/main/java/com/tormentnexus/codex/GhostTextManager.java
package com.tormentnexus.codex;

public class GhostTextManager {
    public String suggest(String input, int cursorPosition) {
        if (input.startsWith("git c")) return "ommit -m \"update\"";
        if (input.startsWith("mvn c")) return "lean install";
        return "";
    }

    public String render(String input, String suggestion) {
        return input + "\033[90m" + suggestion + "\033[0m";
    }
}